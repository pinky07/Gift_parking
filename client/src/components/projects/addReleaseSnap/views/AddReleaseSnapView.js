import React from 'react';

import Heading from 'grommet/components/Heading';
import Form from 'grommet/components/Form';
import FormFields from 'grommet/components/FormFields';
import FormField from 'grommet/components/FormField';
import TextInput from 'grommet/components/TextInput';
import Box from 'grommet/components/Box';
import Footer from 'grommet/components/Footer';
import Button from 'grommet/components/Button';
import DateTime from 'grommet/components/DateTime';
import Layer from 'grommet/components/Layer';

const AddReleaseSnapView = ({viewModel}) => {
  const {errors} = viewModel;
  const {formCallbacks} = viewModel;
  const {projectName} = viewModel;
  const {releaseName} = viewModel;
  const {releaseDate} = viewModel;

  return (
    <Layer align='center' closer={true} onClose={formCallbacks.onClose}>

      <Box pad={{vertical: 'large', horizontal: 'small'}}>

        <Heading>Add a release snap</Heading>
        <h2>Project: {projectName}</h2>

        <Form compact={false}>
          <FormFields>
            <fieldset>

              <FormField label='Release name or id' htmlFor="releaseNameInput" error={errors.releaseName}>
                <TextInput id='releaseNameInput' onDOMChange={formCallbacks.onNameChange} value={releaseName}/>
              </FormField>

              <FormField label='Date' htmlFor="releaseDate" error={errors.releaseDate}>
                <DateTime id='releaseDate' format='YYYY-MM-DD' value={releaseDate}
                          onChange={formCallbacks.onDateChange}/>
              </FormField>

            </fieldset>
          </FormFields>

          <Footer pad={{"vertical": "medium"}}>
            <Box>
              <Button label='Add' type='submit' primary={true} onClick={formCallbacks.onClick}/>
            </Box>
            <Box pad={{horizontal: 'small'}}>
              <Button label='Cancel' secondary={true} onClick={formCallbacks.onClose}/>
            </Box>
          </Footer>
        </Form>
      </Box>
    </Layer>
  );
};

export default AddReleaseSnapView;