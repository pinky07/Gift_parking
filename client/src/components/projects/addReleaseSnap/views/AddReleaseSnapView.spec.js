import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import AddReleaseSnapView from './AddReleaseSnapView';

describe('AddReleaseSnapView', () => {

    it('should render the correct data values and validation errors', () => {
        const viewModel = getViewModelWithErrors();
        const renderer = ShallowRenderer.createRenderer();
        const actual = renderer.render(<AddReleaseSnapView viewModel={viewModel} />);

        expect(actual).toMatchSnapshot();
    });

    function getViewModelWithErrors() {
        return {
            projectName: 'Exceptional Project',
            releaseName: 'Version 4',
            releaseDate: '2017-01-02',
            errors: {
                releaseName: 'error releaseName',
                releaseDate: 'error startDate'
            },
            formCallbacks: {
                onNameChange: jest.fn(),
                onDateChange: jest.fn(),
                onClick: jest.fn(),
                onClose: jest.fn()
            }
        };
    }
});