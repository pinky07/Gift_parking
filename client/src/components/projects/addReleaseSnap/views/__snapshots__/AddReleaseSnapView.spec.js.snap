// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReleaseSnapView should render the correct data values and validation errors 1`] = `
<Layer
  align="center"
  closer={true}
  onClose={[Function]}
>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "small",
        "vertical": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Heading
      tag="h1"
    >
      Add a release snap
    </Heading>
    <h2>
      Project: 
      Exceptional Project
    </h2>
    <Form
      compact={false}
      fill={false}
      pad="none"
    >
      <FormFields>
        <fieldset>
          <FormField
            error="error releaseName"
            htmlFor="releaseNameInput"
            label="Release name or id"
            size="medium"
            strong={false}
          >
            <TextInput
              id="releaseNameInput"
              onDOMChange={[Function]}
              value="Version 4"
            />
          </FormField>
          <FormField
            error="error startDate"
            htmlFor="releaseDate"
            label="Date"
            size="medium"
            strong={false}
          >
            <DateTime
              format="YYYY-MM-DD"
              id="releaseDate"
              onChange={[Function]}
              step={1}
              value="2017-01-02"
            />
          </FormField>
        </fieldset>
      </FormFields>
      <Footer
        align="center"
        direction="row"
        pad={
          Object {
            "vertical": "medium",
          }
        }
        responsive={false}
      >
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad="none"
          responsive={true}
          tag="div"
        >
          <Button
            label="Add"
            method="push"
            onClick={[Function]}
            primary={true}
            type="submit"
          />
        </Box>
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad={
            Object {
              "horizontal": "small",
            }
          }
          responsive={true}
          tag="div"
        >
          <Button
            label="Cancel"
            method="push"
            onClick={[Function]}
            secondary={true}
            type="button"
          />
        </Box>
      </Footer>
    </Form>
  </Box>
</Layer>
`;
